engine/action_transaction.go

```diff
@@ -380,41 +354,40 @@ func (c *Client) RevertTransaction(ctx context.Context, id string) error {
 	transaction.XpubOutputValue = XpubOutputValue{"reverted": 0}
 	transaction.DeletedAt.Valid = true
 	transaction.DeletedAt.Time = time.Now()
+	transaction.TxStatus = TxStatusReverted
 
 	err = transaction.Save(ctx) // update existing record
 
 	return err
 }
 
-// UpdateTransaction will update the broadcast callback transaction info, like: block height, block hash, status, bump.
-func (c *Client) UpdateTransaction(ctx context.Context, callbackResp *broadcast.SubmittedTx) error {
+// HandleTxCallback will update the broadcast callback transaction info, like: block height, block hash, status, bump.
+func (c *Client) HandleTxCallback(ctx context.Context, callbackResp *broadcast.SubmittedTx) error {
+	logger := c.options.logger
 	bump, err := bc.NewBUMPFromStr(callbackResp.MerklePath)
 	if err != nil {
-		c.options.logger.Err(err).Msgf("failed to parse merkle path from broadcast callback - tx: %v", callbackResp)
+		logger.Err(err).Msgf("failed to parse merkle path from broadcast callback - tx: %v", callbackResp)
 		return spverrors.Wrapf(err, "failed to parse merkle path from broadcast callback - tx: %v", callbackResp)
 	}
 
-	txInfo := &chainstate.TransactionInfo{
-		BlockHash:   callbackResp.BlockHash,
-		BlockHeight: callbackResp.BlockHeight,
-		ID:          callbackResp.TxID,
-		TxStatus:    callbackResp.TxStatus,
-		BUMP:        bump,
-	}
+	txID := callbackResp.TxID
 
-	tx, err := c.GetTransaction(ctx, "", txInfo.ID)
+	tx, err := c.GetTransaction(ctx, "", txID)
 	if err != nil {
-		c.options.logger.Err(err).Msgf("failed to get transaction by id: %v", txInfo.ID)
+		logger.Warn().Err(err).Msgf("failed to get transaction by id: %v", txID)
 		return err
 	}
 
-	syncTx, err := GetSyncTransactionByTxID(ctx, txInfo.ID, c.DefaultModelOptions()...)
-	if err != nil {
-		c.options.logger.Err(err).Msgf("failed to get sync transaction by tx id: %v", txInfo.ID)
+	tx.BlockHash = callbackResp.BlockHash
+	tx.BlockHeight = uint64(callbackResp.BlockHeight)
+	tx.SetBUMP(bump)
+	tx.UpdateFromBroadcastStatus(callbackResp.TxStatus)
+
+	if err := tx.Save(ctx); err != nil {
```

Comment:
maybe is a good idea to log this error `logger.Err(err).Msgf("failed to save transaction - tx: %v", txID)`

Reply:
The error is actually is logged in `actions::transactions::broadcastCallback`.

Generally, I'm a fan of the approach where logging is performed at the highest possible level. For internal functions, I prefer to return errors (possibly with wrapping) and perform the logging at the topmost caller. This prevents logging an error multiple times and provides more business context about what happened.
```golang
func level3() error {
   return someErrorOnLevel3
}

func level2() error {
   if err := level3(); err != nil {
       return someErrorOnLevel2.Wrap(err)
   }
   // ...
}

// Base level
func level1() {
   if err := level2(); err != nil {
      log.Err(err)
      doSomeAction("For requests, it would return a failed status; for models, it would change some fields in the struct")
   }
}
```